hubconf.py

```diff
@@ -24,78 +24,83 @@
 _MODEL_URL = "https://github.com/castorini/howl-models/archive/v1.0.0.zip"
 _MODEL_CACHE_FOLDER = "howl-models"
 
-@dataclasses.dataclass
-class _ModelSettings:
-    path: str # Relative to the howl-models root folder
-    model_name: str
-    vocab: typing.List[str]
-    inference_sequence: typing.List[int]
-    token_type: str
-    use_frame: bool
+
+def hey_fire_fox(pretrained=True, **kwargs):
+    """Pretrained model for Firefox Voice"""
+    engine, ctx = _load_model(pretrained, "howl/hey-fire-fox", **kwargs)
+    return engine, ctx
 
 
-def hey_fire_fox(pretrained=True, **kwargs) \
+def _load_model(pretrained: bool, workspace_path: str, **kwargs) \
         -> typing.Tuple[inference.InferenceEngine, context.InferenceContext]:
     """
-    Pretrained model for Firefox Voice
+    Loads howl model from a workspace
+
+    Arguments:
+        pretrained (bool): load pretrained model weights
+        workspace_path (str): relative path to workspace from root of howl-models
+
+    Returns the inference engine and context
     """
-    # Audio inference settings
-    conf = _ModelSettings(path="howl/hey-fire-fox",
-                          model_name="res8",
-                          vocab=["hey", "fire", "fox"],
-                          inference_sequence=[0, 1, 2],
-                          token_type="word",
-                          use_frame=True)
-    ats = transform.augment.AudioTransformSettings(num_mels=40)
-    _SETTINGS._training = settings.TrainingSettings(max_window_size_seconds=0.5, eval_stride_size_seconds=0.15)
 
     # Separate `reload_models` flag since PyTorch will pop the 'force_reload' flag
     reload_models = kwargs.pop('reload_models', False)
-    cached_folder =_download_howl_models(reload_models)
+    cached_folder = _download_howl_models(reload_models)
+    workspace_path = pathlib.Path(cached_folder) / workspace_path
+    ws = howl_model.Workspace(workspace_path, delete_existing=False)
 
-    # Load workspace from path
-    ws_path: pathlib.Path = pathlib.Path(cached_folder) / conf.path
-    ws = howl_model.Workspace(ws_path, delete_existing=False)
+    # Load model settings
+    ws_settings = ws.load_settings()
+    model_settings = ws_settings.model
+    audio_transform_settings = transform.augment.AudioTransformSettings(**ws_settings.audio_transform)
+    _SETTINGS._training = settings.TrainingSettings(**ws_settings.training)
```

Comment:
can this be loaded to load_settings function of WorkSpace?

Reply:
Done